view/frontend/templates/js/replacejs.phtml

```diff
@@ -753,85 +755,56 @@ ob_start();
                         var BC = BoltCheckout.configure(cart, hintBarrier.promise, callbacks);
                 });
             } else {
-            // define the params sent with the request variable
-            var params = [];
-            params.push('form_key=' + $('[name="form_key"]').val());
-            params = params.join('&');
-
             var hintBarrier = boltBarrier();
+            var cartBarrier = boltBarrier();
 
-            cart = new Promise(function (resolve, reject) {
-
-                createRequest = true;
-
-                // send create order request
-                $.get(settings.create_order_url, params)
-                    .done(function (data) {
+            customerData.get('boltcart').subscribe(function(data) {
+                cart = data.cart !== undefined ? data.cart : {};
+                if (JSON.stringify(cart) === oldBoltCartValue) {
+                    return;
+                }
 
                         var boltRestricted = !!data.restrict;
 
                         toggleCheckoutIfNeeded(boltRestricted);
 
                         // Stop if Bolt checkout is restricted
                         if (boltRestricted) {
-                            if (popUpOpen) reject(new Error(data.message));
-                            hintBarrier.resolve(hints);
-                            return;
-                        }
-
-                        if (data.status !== 'success') {
-                            reject(new Error(data.message || 'Network request failed'));
-                            hintBarrier.resolve(hints);
-                            return;
-                        }
-
-                        if (!data.cart) {
-                            reject(new Error('The cart info is missing.'));
-                            hintBarrier.resolve(hints);
-                            return;
-                        }
-
-                        if (!data.cart.orderToken) {
-                            reject(new Error('The cart is empty.'));
-                            hintBarrier.resolve(hints);
-                            return;
-                        }
+                            cart = {};
+                            hints = {};
```

Comment:
let's just return here and remove else below. else block below is a bit too broated https://www.itamarweiss.com/personal/2018/02/28/return-early-pattern.html

Reply:
Looks like we can't just return. Only lines 777-780 is in else block. Even if bolt restricted we should resolve promises or reconfigure.
Let me handle it in next stage of refactoring: we need to extract function that just calculates cart and hints. In this case, we are able to do early return.